accounts-db/src/account_locks.rs

```diff
@@ -21,10 +21,65 @@ impl AccountLocks {
     /// The bool in the tuple indicates if the account is writable.
     /// Returns an error if any of the accounts are already locked in a way
     /// that conflicts with the requested lock.
+    /// NOTE this is the pre-SIMD83 logic and can be removed once SIMD83 is active.
     pub fn try_lock_accounts<'a>(
         &mut self,
         keys: impl Iterator<Item = (&'a Pubkey, bool)> + Clone,
-    ) -> Result<(), TransactionError> {
+    ) -> Result<()> {
+        self.can_lock_accounts(keys.clone())?;
+        self.lock_accounts(keys);
+
+        Ok(())
+    }
+
+    /// Lock accounts for all transactions in a batch which don't conflict
+    /// with existing locks. Returns a vector of `TransactionResult` indicating
+    /// success or failure for each transaction in the batch.
+    /// NOTE this is the SIMD83 logic; after the feature is active, it becomes
+    /// the only logic, and this note can be removed with the feature gate.
+    pub fn try_lock_transaction_batch<'a>(
+        &mut self,
+        validated_batch_keys: Vec<Result<impl Iterator<Item = (&'a Pubkey, bool)> + Clone>>,
+    ) -> Vec<Result<()>> {
+        // HANA TODO the vec allocation here is unfortunate but hard to avoid
+        // we cannot do this in one closure because of borrow rules
+        // play around with alternate strategies, according to benches this may be up to
+        // 50% slower for small batches and few locks, but for large batches and many locks
```

Comment:
bench using jemalloc? i'd think it would do a reasonable job of just keeping the mem in thread-local cache for re-use